File: 0001_0100/0036/0036.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    n = int(f.readline())
    p = [0, 1] * n
    i = 3
    while i * i < 2 * n:
        if p[i]:
            for j in range(i * i, 2 * n, 2 * i):
                p[j] = 0
        i += 2
    q.write(str(sum(p[n + 1 :])))

File: 0001_0100/0036/0036.go
package main

import (
	"fmt"
	"os"
)

func main() {
	f, _ := os.Open("input.txt")
	q, _ := os.Create("output.txt")
	var n int
	fmt.Fscan(f, &n)
	p := make([]bool, 2*n)
	for i := 2; i < n; i++ {
		if !p[i] {
			for j := 2 * i; j < 2*n; j += i {
				p[j] = true
			}
		}
	}
	k := 0
	for i := n + 1; i < 2*n; i++ {
		if !p[i] {
			k++
		}
	}
	fmt.Fprint(q, k)
}
